types/d3-selection/index.d.ts

```diff
@@ -176,7 +177,7 @@ export interface Selection<GElement extends BaseType, Datum, PElement extends Ba
      *
      * @param selector CSS selector string
      */
-    select<DescElement extends BaseType>(selector: string): Selection<DescElement, Datum, PElement, PDatum>;
+    select<DescElement extends BaseType>(selector: string): Selection<DescElement, Datum, PElement | HTMLElement, PDatum | any>;
```

Comment:
This change worries me, for two reasons:

1. For any type T, `T | any` is simplified by the compiler to `any`. This winds up defeating any type safety we have over a parent element's data.

    ![image](https://user-images.githubusercontent.com/194893/28167121-6c37d2a4-67a8-11e7-9088-6f0c2998a01b.png)

2. Is it a property of `ShadowRoot`s that the parent element is an HTML element? As written, this change will ensure that we can coerce any parent element to an HTML element, even if we're working with SVG elements.

It seems like there are some subtleties around `ShadowRoot` that we should be careful around, rather than accidentally blowing holes in the type definitions. Thoughts?